tests/test_omegaconf.py

```diff
@@ -518,3 +518,93 @@ def test_resolve(cfg: Any, expected: Any) -> None:
 def test_resolve_invalid_input() -> None:
     with raises(ValueError):
         OmegaConf.resolve("aaa")  # type: ignore
+
+
+@mark.parametrize(
+    ("cfg_params,clear_resolver_params,expected"),
+    [
+        # tuple
+        (
+            # cfg_params
+            dict(
+                name="TR1",
+                resolver=lambda x: str(x).lower(),
+                use_cache=False,
+                replace=True,
+            ),
+            # clear_resolver_params
+            dict(name="TR1"),
+            # expected
+            dict(pre_addition=False, post_addition=True, post_removal=False),
+        ),
+        # tuple
+        (
+            # cfg_params
+            dict(
+                name="TR2",
+                resolver=lambda x: str(x).upper(),
+                use_cache=False,
+                replace=True,
+            ),
+            # clear_resolver_params
+            dict(name="TR2"),
+            # expected
+            dict(pre_addition=False, post_addition=True, post_removal=False),
+        ),
+        # tuple
+        (
+            # cfg_params
+            dict(
+                name="TR3",
+                resolver=lambda x: str(x).upper(),
+                use_cache=False,
+                replace=False,
+            ),
+            # clear_resolver_params
+            dict(name="TR3"),
+            # expected
+            dict(pre_addition=False, post_addition=True, post_removal=False),
+        ),
+        # tuple
+        (
+            # cfg_params: Check with a default resolver
+            dict(),
+            # clear_resolver_params
+            dict(name="oc.env"),
+            # expected
+            dict(pre_addition=True, post_addition=True, post_removal=False),
+        ),
+        # tuple
+        (
+            # cfg_params: Check with a default resolver
+            dict(),
+            # clear_resolver_params
+            dict(name="oc.create"),
+            # expected
+            dict(pre_addition=True, post_addition=True, post_removal=False),
+        ),
+        # tuple
+        (
+            # cfg_params: Check with a nonexisting resolver
+            dict(),
+            # clear_resolver_params
+            dict(name="abcdef"),
+            # expected
+            dict(pre_addition=False, post_addition=False, post_removal=False),
+        ),
+    ],
+)
+def test_clear_resolver(
+    cfg_params: Any, clear_resolver_params: Any, expected: Any
+) -> None:
+
+    OmegaConf.clear_resolvers()
+    name = clear_resolver_params.get("name")
+    assert expected["pre_addition"] == OmegaConf.has_resolver(name)
+    if cfg_params:
+        OmegaConf.register_new_resolver(**cfg_params)
+        assert expected["post_addition"] == OmegaConf.has_resolver(name)
+
+    assert OmegaConf.clear_resolver(**clear_resolver_params)
```

Comment:
Okay. Makes sense. However, I thought @omry suggested the bool output for True when resolver is removed (*if something has been removed, then it does not exist anymore; and it’s current state is the same as something that never existed*) and False when not removed for whatever reason.

Reply:
I guess my point is that it is not very useful to return a bool 
> output of `True` when resolver is removed

Agreed.

 > or for when the resolver did not exist in the first place

I think it would be more useful to output `False` if the resolver did not exist in the first place.
The reason is that returning `True` in all cases (which is what the function does currently) is not very useful.

Reply:
From Omry's [original comment](https://github.com/omry/omegaconf/issues/769#issuecomment-884302355) about returning a bool:

> A simple API for clear resolver can be:
> 
> ```python
> def clear_resolver(name: str) -> bool:
>   ...
> ```
> 
> This will remove a resolver and return False if it's not found, allowing the user to handle this case.
> It will allow a user to remove all resolvers, including built in ones. I think this is okay.

In particular, "return False if it's not found"

Reply:
@Jasha10 Thank you, for pointing it out. I shouldn't have missed it. I will make the change.